pkg/container-runtime/configurator.go

```diff
@@ -0,0 +1,305 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package container_runtime
+
+import (
+	"context"
+	"fmt"
+	"github.com/sealerio/sealer/pkg/infradriver"
+	"github.com/sealerio/sealer/utils/os"
+	"golang.org/x/sync/errgroup"
+	"net"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	RemoteAddEtcHosts     = "cat /etc/hosts |grep '%s' || echo '%s' >> /etc/hosts"
+	DockerCertDir         = "/etc/docker/certs.d"
+	ContainerdCertDir     = "/etc/containerd/certs.d"
+	RestartServiceCommand = "systemctl restart %s"
+	DockerLoginCommand    = "docker login -u %s -p %s %s && " + KubeletAuthCommand
+	NerdctlLoginCommand   = "nerdctl login -u %s -p %s %s && " + KubeletAuthCommand
+	KubeletAuthCommand    = "mkdir -p /var/lib/kubelet && cp /root/.docker/config.json /var/lib/kubelet"
+	DockerDaemonFile      = "/etc/docker/daemon.json"
+	DefaultEndpoint       = "sea.hub:5000"
+	DumpConfigContainerd  = "containerd --config %s config dump > /etc/containerd/config.toml"
+	ModifyFileCommand     = "sed -i \"s/%s/%s/g\" %s"
+)
+
+//Configurator provide configuration Interface for different container runtime.
+type Configurator interface {
+	// ConfigRegistry config registry via each container runtime ip address
+	ConfigRegistry(RegistryConfig, []net.IP) error
+}
+
+type RegistryConfig struct {
+	SkipTLSVerify      bool
+	RegistryDeployHost string
+	Endpoint           string
+	Domain             string
+	Port               string
+	Username           string
+	Password           string
+	CaFile             string
+	KeyFile            string
+	CertFile           string
+}
+
+type DockerRuntimeConfigurator struct {
```

Comment:
We always input all Host IPs into every DockerRuntimeConfigurator's function, so I suggest that we encapsulate all host IPs as a field in DockerRuntimeConfigurator's definition.